internal/server/http/server.go

```diff
@@ -73,31 +72,30 @@ func PrepareAccessControl(config killgrave.ConfigCORS) (h []handlers.CORSOption)
 // Build read all the files on the impostersPath and add different
 // handlers for each imposter
 func (s *Server) Build() error {
-	if _, err := os.Stat(s.impostersPath); os.IsNotExist(err) {
-		return fmt.Errorf("%w: the directory %s doesn't exists", err, s.impostersPath)
+	done := make(chan struct{})
+	imposterFiles, errc := findImposters(s.impostersPath, done)
+	go s.processImposters(imposterFiles, done)
+
+	if err := <-errc; err != nil {
+		close(done)
+		return fmt.Errorf("%w: failed to find imposters", err)
 	}
-	var imposterFileCh = make(chan string)
-	var done = make(chan bool)
+	return nil
+}
 
-	go func() {
-		findImposters(s.impostersPath, imposterFileCh)
-		done <- true
-	}()
-	for {
+func (s *Server) processImposters(imposterFiles <-chan string, done <-chan struct{}) {
+	var imposters []Imposter
+	for f := range imposterFiles {
 		select {
-		case f := <-imposterFileCh:
-			var imposters []Imposter
-			err := s.unmarshalImposters(f, &imposters)
-			if err != nil {
+		case <-done:
+			return
```

Comment:
Checking the branch I see that there no test that pass for here, can you add it?

Reply:
I have a bit of a problem here. It will only get to this branch if `filepath.Walk` would return an error on one of the file/directory but be successful on others. I was playing with creating files with bad permissions but no success yet, maybe you have an idea how to test that? The other idea would be to create test cases for not exported functions, how do feel about that?